src/aind_metadata_service/sharepoint/client.py

```diff
@@ -3,150 +3,244 @@
+    """This class contains the API to connect to Sharepoint Database."""
+
+    # Microsoft Graph API Default values
+    GRAPH_API_URL = "https://graph.microsoft.com/v1.0"
+    SCOPE = "https://graph.microsoft.com/.default"
 
     def __init__(
         self,
-        nsb_site_url: str,
-        las_site_url: str,
+        aind_site_id: str,
+        las_site_id: str,
+        nsb_2019_list_id: str,
+        nsb_2023_list_id: str,
+        las_2020_list_id: str,
         client_id: str,
-        client_secret: str,
-        nsb_2019_list_title: str,
-        nsb_2023_list_title: str,
-        las_2020_list_title: str,
+        client_secret: SecretStr,
+        tenant_id: str,
     ) -> None:
         """
-        Initialize a client
+        Initailize the SharePointClient with the required parameters.
         Parameters
         ----------
-        nsb_site_url : str
-           sharepoint site url for nsb procedures
-        las_site_url : str
-           sharepoint site url for las procedures
+        aind_site_id : str
+            Sharepoint Site ID for AIND Domain
+        las_site_id : str
+            Sharepoint Site ID for LAS Domain
+        nsb_2019_list_id : str
+            List ID for NSB 2019 procedures
+        nsb_2023_list_id : str
+            List ID for NSB 2023 procedures
+        las_2020_list_id : str
+            List ID for LAS 2020 procedures
         client_id : str
-            username for principal account to access sharepoint
-        client_secret : str
-            password for principal account to access sharepoint
-        nsb_2019_list_title : str
-            Title for nsb 2019 list
-        nsb_2023_list_title : str
-            Title for nsb 2023 list
-        las_2020_list_title : str
-            Title for las 2020 list
+            Client ID for the principal account
+        client_secret : SecretStr
+            Client Secret for the principal account
+        tenant_id : str
+            Tenant ID for the principal account
         """
+        self.aind_site_id = aind_site_id
+        self.las_site_id = las_site_id
+        self.nsb_2019_list_id = nsb_2019_list_id
+        self.nsb_2023_list_id = nsb_2023_list_id
+        self.las_2020_list_id = las_2020_list_id
         self.client_id = client_id
         self.client_secret = client_secret
-        self.credentials = ClientCredential(self.client_id, self.client_secret)
-        self.nsb_site_url = nsb_site_url
-        self.las_site_url = las_site_url
-        self.nsb_2019_list_title = nsb_2019_list_title
-        self.nsb_2023_list_title = nsb_2023_list_title
-        self.las_2020_list_title = las_2020_list_title
-
-    def get_client_context(self, site_url):
-        """Construct client context with principal account"""
-        return ClientContext(site_url).with_credentials(self.credentials)
+        self.tenant_id = tenant_id
+        self.token_url = (
+            f"https://login.microsoftonline.com/{self.tenant_id}/"
```

Comment:
You should move the host to the config file

Reply:
I moved all the configs to the SharepointSettings class with default values. The `token_url` is filled in using a field_validator if it's supplied in the config file